src/Tools/dotnet-counters/Exporters/ConsoleWriter.cs

```diff
@@ -150,31 +150,43 @@ public void AssignRowsAndInitializeDisplay()
             {
                 Console.WriteLine($"[{provider.Name}]"); row++;
 
+                foreach (ObservedCounter counter in provider.Counters.Values)
+                {
+                    counter.Row = -1;
+                    if (counter.RenderValueInline)
+                    {
+                        foreach (ObservedTagSet tagSet in counter.TagSets.Values)
+                        {
+                            tagSet.Row = -1;
+                        }
+                    }
+                }
+
                 foreach (ObservedCounter counter in provider.Counters.Values.OrderBy(c => c.DisplayName))
                 {
 
                     string name = MakeFixedWidth($"{new string(' ', Indent)}{counter.DisplayName}", Indent + _maxNameLength);
+                    if (row >= _consoleHeight + _topRow - 1) // prevents from displaying more counters than vertical space available
+                    {
+                        break;
+                    }
                     counter.Row = row++;
                     if (counter.RenderValueInline)
                     {
-                        if (row >= _consoleHeight) // prevents from displaying more counters than vertical space available
-                        {
-                            break;
-                        }
                         Console.WriteLine($"{name} {FormatValue(counter.LastValue)}");
                     }
                     else
                     {
                         Console.WriteLine(name);
                         foreach (ObservedTagSet tagSet in counter.TagSets.Values.OrderBy(t => t.Tags))
                         {
-                            if (row >= _consoleHeight)
+                            string tagName = MakeFixedWidth($"{new string(' ', 2 * Indent)}{tagSet.Tags}", Indent + _maxNameLength);
+                            Console.WriteLine($"{tagName} {FormatValue(tagSet.LastValue)}");
+
+                            if (row >= _consoleHeight + _topRow - 1)
```

Comment:
In the beginning `row == _topRow`. That means that this is going to be equivalent to: increments to `row` + 1 has to be greater than `_consoleHeight`. That feels like it should be enough to solve the issue. Don't mind the row=-1 part for verifiability's sake. However. it does feel that if it's rendered, a console line num should be assigned, right? This breaks before it has the chance to.

Reply:
You are right - I have pushed a fix.  Also added support for utilizing all of the console buffer, not just height-1.